src/whitenoise/database/accounts_groups.rs

```diff
@@ -0,0 +1,462 @@
+                // Fetch the existing record
+                let existing =
+                    Self::find_by_account_and_group(account_pubkey, mls_group_id, database)
+                        .await?
+                        .ok_or(sqlx::Error::RowNotFound)?;
+                Ok((existing, false))
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Finds all visible AccountGroups for a given account.
+    /// Visible means: user_confirmation is NULL (pending) or true (accepted).
+    /// Declined groups (user_confirmation = false) are hidden.
+    pub(crate) async fn find_visible_for_account(
+        account_pubkey: &PublicKey,
+        database: &Database,
+    ) -> Result<Vec<Self>, sqlx::Error> {
+        let rows = sqlx::query_as::<_, AccountGroupRow>(
+            "SELECT id, account_pubkey, mls_group_id, user_confirmation, created_at, updated_at
+             FROM accounts_groups
+             WHERE account_pubkey = ? AND (user_confirmation IS NULL OR user_confirmation = 1)",
+        )
+        .bind(account_pubkey.to_hex())
+        .fetch_all(&database.pool)
+        .await?;
+
+        Ok(rows.into_iter().map(|r| r.into()).collect())
+    }
+
+    /// Finds all pending AccountGroups for a given account.
+    /// Pending means: user_confirmation is NULL.
+    pub(crate) async fn find_pending_for_account(
+        account_pubkey: &PublicKey,
+        database: &Database,
+    ) -> Result<Vec<Self>, sqlx::Error> {
+        let rows = sqlx::query_as::<_, AccountGroupRow>(
+            "SELECT id, account_pubkey, mls_group_id, user_confirmation, created_at, updated_at
+             FROM accounts_groups
+             WHERE account_pubkey = ? AND user_confirmation IS NULL",
+        )
+        .bind(account_pubkey.to_hex())
+        .fetch_all(&database.pool)
+        .await?;
+
+        Ok(rows.into_iter().map(|r| r.into()).collect())
+    }
+
+    /// Updates the user_confirmation status for this AccountGroup.
+    pub(crate) async fn update_user_confirmation(
+        &self,
+        user_confirmation: bool,
+        database: &Database,
+    ) -> Result<Self, sqlx::Error> {
+        let id = self.id.ok_or_else(|| {
+            sqlx::Error::Protocol("Cannot update unsaved AccountGroup".to_string())
+        })?;
+
+        let now_ms = Utc::now().timestamp_millis();
+        let confirmation_int: i64 = if user_confirmation { 1 } else { 0 };
+
+        let row = sqlx::query_as::<_, AccountGroupRow>(
+            "UPDATE accounts_groups
+             SET user_confirmation = ?, updated_at = ?
+             WHERE id = ?
+             RETURNING id, account_pubkey, mls_group_id, user_confirmation, created_at, updated_at",
+        )
+        .bind(confirmation_int)
+        .bind(now_ms)
+        .bind(id)
+        .fetch_one(&database.pool)
+        .await?;
+
+        Ok(row.into())
+    }
+
+    /// Deletes this AccountGroup from the database.
+    #[allow(dead_code)]
```

Comment:
If this is dead code perhaps we should remove the method for now until we actually use it somewhere?